Bottom: 1da40b4e76554279395a47fec49f8fab4b61b6ea
Top:    1c8a61ce3fc237dcb49d8b0e07f92b9be2c9ae79
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-03-20 13:51:34 +0800

tun: introduce tap_get_skb_array()

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index c418f0a..ff14c99 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -2613,6 +2613,19 @@ struct socket *tun_get_socket(struct file *file)
 }
 EXPORT_SYMBOL_GPL(tun_get_socket);
 
+struct skb_array *tap_get_skb_array(struct file *file)
+{
+       struct tun_file *tfile;
+
+       if (file->f_op != &tun_fops)
+               return ERR_PTR(-EINVAL);
+       tfile = file->private_data;
+       if (!tfile)
+               return ERR_PTR(-EBADFD);
+       return &tfile->tx_array;
+}
+EXPORT_SYMBOL_GPL(tap_get_skb_array);
+
 module_init(tun_init);
 module_exit(tun_cleanup);
 MODULE_DESCRIPTION(DRV_DESCRIPTION);
